### PR TITLE
add css rule to keep header visible when maxHeight is set

### LINKED ISSUE #3159: Table header in TableView using asyncList is not visible
# 🐛 Bug Report 

In a TableView that uses an async list as a data source with a `maxHeight` prop defined, the table header is not visible.

## 🤔 Expected Behavior

When a TableView is rendered using an asynclist as datasource with a `maxHeight` prop defined, the table headers should be visible.

## 😯 Current Behavior

Currently, the table headers are not visible: https://codesandbox.io/s/exciting-gates-vurwv0

## 💁 Possible Solution

@snowystinger suggested on Slack that a flex rule be added to the `.spectrum-Table-headWrapper` class, which seemed to resolve the issue for me when applied via devtools. Will submit a PR with this fix.

## 🔦 Context

We're using async tables to display data that needs column headers for legibility and to be sortable. We are setting a maxHeight on the TableView in order to make use of the infinite scrolling function, because the height of the TableView's container is not fixed.

## 💻 Code Sample

https://codesandbox.io/s/exciting-gates-vurwv0

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.17.0
| Browser          | Vivaldi 5.2.2623.48 (Stable channel) (x86_64) 
| Operating System | macOS 12.3.1

## 🧢 Your Company/Team

Adobe/Marketo Customer Engineering

### PR DESCRIPTION
**Pull request by ihgrant**
Closes #3159 

I added a Chromatic story to illustrate the fix - haven't used Chromatic or Storybook before so let me know if I did something wrong!

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

* on this branch, run the storybook and modify `packages/@react-spectrum/table/stories/Table.stories.tsx` on line 157 to change the `height` prop to `maxHeight`.
* verify that the header is fully visible in the "dynamic" TableView story

### CODE REVIEW

File: packages/@react-spectrum/table/test/Table.test.js [hunk 1]
```diff
@@ -120,7 +120,10 @@ describe('TableView', function () {
   });
 
   beforeEach(() => {
-    jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => cb());
+    jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => {
```
Comment (snowystinger) [inline packages/@react-spectrum/table/test/Table.test.js:123]: can we revert this change?

File: packages/@react-spectrum/table/chromatic/TableView.chromatic.tsx [hunk 2]
```diff
@@ -198,6 +198,10 @@ export const NestedColumns = Template().bind({});
 NestedColumns.storyName = 'nested columns';
 NestedColumns.args = {columns: nestedColumns, items};
 
+export const MaxHeight = Template.bind({});
```
Comment (snowystinger) [inline packages/@react-spectrum/table/chromatic/TableView.chromatic.tsx:201]: So this actually doesn't render anything
you can run our chromatic storybook if you run the command `yarn start:chromatic` as defined in our package.json

That aside, I don't think we really need this, we'd really need to test it against the async version to meet your reproduction from what you've said. It also looks like the Template is not applying the maxHeight where you expect.

I think it's plenty this time to just contribute the fix.

**Review by snowystinger — COMMENTED** [review]
Thanks for submitting a fix! couple small notes

heads up, I did merge in main

File: packages/@react-spectrum/table/test/Table.test.js (hunk 1 above)
Comment (ihgrant) [reply packages/@react-spectrum/table/test/Table.test.js:123]: Yeah, will do.

File: packages/@react-spectrum/table/chromatic/TableView.chromatic.tsx (hunk 2 above)
Comment (ihgrant) [reply packages/@react-spectrum/table/chromatic/TableView.chromatic.tsx:201]: Okay, despite my initial report I *think* this bug occurs even with non-async lists, so I'll spend a bit of time to see if I can get this working. I wasn't aware I could run Chromatic locally without a token, that should help.

> So this actually doesn't render anything

Is this because I am (mistakenly) not calling Template?

Comment (snowystinger) [reply packages/@react-spectrum/table/chromatic/TableView.chromatic.tsx:201]: Honestly, I didn't dig into why it wasn't rendering anything. If it happens in the non-async case, that'd be immensely helpful. Is it easy to reproduce?
I'd check the definition of Template a little further up in the file, looks like it already has a maxHeight on the TableView. You also don't need to use the Template, you can just return a component. So if you have a reproduction case, you could just copy it in and it should work.
If this is taking too much time, I'm really ok with just the change.

Comment (ihgrant) [reply packages/@react-spectrum/table/chromatic/TableView.chromatic.tsx]: Got the Chromatic test working using a literal component which demonstrates the bug if the CSS line is commented out. Thanks for the pointers!

**Review by snowystinger — COMMENTED** [review]
Awesome, chromatic looks good, so it's just undoing the changes in the test file, then I can approve

Thanks!!

**Review by snowystinger — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]
LGTM, tested the css change locally w/ the chromatic story. Thanks for the fix!
